sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/AbstractLNAdapter.java

```diff
@@ -574,45 +531,72 @@ public TExtRef checkExtRefInfoCoherence(@NonNull ExtRefInfo extRefInfo) throws S
         String binderLnPrefix = bindingInfo.getPrefix();
 
         IEDAdapter binderIEDAdapter;
-        if(!binderIedName.equals( getCurrentIed().getName())){ // external binding
+
+        if (!binderIedName.equals(getCurrentIed().getName())) {
             binderIEDAdapter = getCurrentScd().getIEDAdapterByName(binderIedName);
         } else {
             binderIEDAdapter = getCurrentIed();
         }
-        LDeviceAdapter binderLDAdapter = binderIEDAdapter.findLDeviceAdapterByLdInst(binderLdInst)
+        LDeviceAdapter binderLDeviceAdapter = binderIEDAdapter.findLDeviceAdapterByLdInst(binderLdInst)
                 .orElseThrow(
                         () -> new ScdException(
                                 String.format("Unknown LDevice (%s) in IED (%s)", binderLdInst, binderIedName)
                         )
                 );
-        AbstractLNAdapter<?> anLNAdapter = AbstractLNAdapter.builder()
-                .withLDeviceAdapter(binderLDAdapter)
-                .withLnClass(binderLnClass)
-                .withLnInst(binderLnInst)
-                .withLnPrefix(binderLnPrefix)
-                .build();
-
-        boolean isCoherent;
 
         ExtRefSourceInfo sourceInfo = extRefInfo.getSourceInfo();
-        if (sourceInfo == null || sourceInfo.isNull()) { // to be refined : what to do here functionally ?
-            return extRef;
-        }
-        List<ControlBlock<?>> cbs = anLNAdapter.getControlSetByExtRefInfo(extRefInfo);
-        isCoherent = !cbs.isEmpty() && cbs.stream()
-                .anyMatch(controlBlock -> controlBlock.getName().equals(sourceInfo.getSrcCBName()));
-        if(!isCoherent){
-            String msg = String.format("Unknown control block %s in %s%s/%s%s%s",
-                    sourceInfo.getSrcCBName(), binderIedName,binderLdInst,
-                    binderLnPrefix , binderLnClass, binderLnInst);
+        List<AbstractLNAdapter<?>> aLNAdapters = new ArrayList<>();
```

Comment:
```java
        List<AbstractLNAdapter<?>> aLNAdapters = new ArrayList<>();
        if(StringUtils.isBlank(sourceInfo.getSrcLNClass())){
            aLNAdapters.addAll(binderLDeviceAdapter.getLNAdaptersInclundigLN0());
        } else {
            if(TLLN0Enum.LLN_0.value().equals(sourceInfo.getSrcLNClass())){
                aLNAdapters.add(binderLDeviceAdapter.getLN0Adapter());
            } else {
                aLNAdapters.add(binderLDeviceAdapter.getLNAdapter(sourceInfo.getSrcLNClass(), sourceInfo.getSrcLNInst(), sourceInfo.getSrcPrefix()));
            }
        }
```
This part can be extracted from `checkExtRefInfoCoherence` as a function that collecting LNAdapter objects for specific LDeviceAdapter and LN attributes (lnClass, inst and prefix)

Reply:
ok